### PR TITLE
font-th-sarabun-new 1.35 (new cask)

### PR DESCRIPTION
**Pull request by phwt**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

Adding **TH Sarabun New**, Thailand's official typeface, note that this one is a different font to the [`font-sarabun`](https://formulae.brew.sh/cask/font-sarabun) cask.

### CODE REVIEW

File: Casks/font/font-t/font-th-sarabun-new.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cask "font-th-sarabun-new" do
+  version "1.0"
+  sha256 :no_check
+
+  url "https://www.f0nt.com/?dl_name=sipafonts/THSarabunNew.zip"
+  name "TH Sarabun New"
+  homepage "https://www.f0nt.com/release/th-sarabun-new/"
```
Comment (bevanjkay) [inline Casks/font/font-t/font-th-sarabun-new.rb:7]: Is this the creator's website?

**Comment by khipp** [comment]
> Adding TH Sarabun New, Thailand's official typeface, note that this one is a different font to the [font-sarabun](https://formulae.brew.sh/cask/font-sarabun) cask.

Could you clarify these differences?

Based on the Google Fonts page, it seems to be the same font:

> Sarabun is an open source multi-script webfont that supports both Latin and Thai. It is the "TH Sarabun New" font, made available under the Open Font License.

File: Casks/font/font-t/font-th-sarabun-new.rb (hunk 1 above)
Comment (phwt) [reply Casks/font/font-t/font-th-sarabun-new.rb:7]: Not directly the creator's site but a site that publishes the fonts selected in the font contest held by SIPA (Software Industry Promotion Agency) and Department of Intellectual Property (DIP) in 2006, there are multiple publishers, but this site is the most popular (and still online) one.

The one directly published by SIPA and DIP are now offline.
- SIPA: http://www.sipa.or.th/main/index.php?task=view&id=659&Itemid=63
- DIP: https://www.ipthailand.org/dip/dmdocuments/dip_sipa_font.zip

**Comment by phwt** [comment]
Hi @khipp, both **TH Sarabun New** (this cask) and **Sarabun** (`font-sarabun` cask) are from the same creator Suppakit Chalermlarp, I'll help highlights the differences below:

- **Distribution**
  - **TH Sararbun New** distributed by SIPA and DIP
  - **Sarabun** distributed by Cadson Demak (typeface and design agency in Thailand)
- **Offered font variations**
  - Below screenshot is the package content (left: TH Sarabun New, right: Sarabun)
  - image
- **Different font weight for the normal/regular version of the font**
  - Preview of `THSarabunNew.ttf`
    - Screenshot 2024-11-16 at 19 43 16
  - Preview of `Sarabun-Regular.ttf`
    - Screenshot 2024-11-16 at 19 43 20

Based on this, the two can be considered a different variation of the same font, the most crucial point is the different regular font weight, which will impact how the document will look like, therefore both cannot be interchangeably used.

Hope this help clarifies the differences!

File: Casks/font/font-t/font-th-sarabun-new.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+cask "font-th-sarabun-new" do
+  version "1.0"
```
Comment (khipp) [inline Casks/font/font-t/font-th-sarabun-new.rb:2]:
```suggestion
  version "1.35"
```

File: Casks/font/font-t/font-th-sarabun-new.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  sha256 :no_check
+
+  url "https://www.f0nt.com/?dl_name=sipafonts/THSarabunNew.zip"
+  name "TH Sarabun New"
+  homepage "https://www.f0nt.com/release/th-sarabun-new/"
+
```
Comment (khipp) [inline Casks/font/font-t/font-th-sarabun-new.rb:8]:
```suggestion

  livecheck do
    url :homepage
    regex(/Sarabun\s+New.*?v?(\d+(?:\.\d+)+)/i)
  end

```

File: Casks/font/font-t/font-th-sarabun-new.rb (hunk 1 above)
Comment (bevanjkay) [reply Casks/font/font-t/font-th-sarabun-new.rb:7]: We tend to only include fonts that are distributed from the original source, not from external repositories.
The exception to this rule is Google Fonts, as the addition and maintenance of these fonts are handled systematically by a script.

**Comment by bevanjkay** [comment]
We tend to only include fonts that are distributed from the original source, not from external repositories.
The exception to this rule is Google Fonts, as the addition and maintenance of these fonts are handled systematically by a script.

In this instance think you'd be best including this in your [own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap), and we not include it in `homebrew-cask`.

**Comment by phwt** [comment]
> We tend to only include fonts that are distributed from the original source, not from external repositories. The exception to this rule is Google Fonts, as the addition and maintenance of these fonts are handled systematically by a script.
> 
> In this instance think you'd be best including this in your [own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap), and we not include it in `homebrew-cask`.

Thanks for the explanation, I understand the maintenance concern, since the original source is no longer available, and the current version (`font-sarabun`) is distinctly different from the version commonly used in Thailand, setting up my own tap would be the only option as you suggested.

**Pull request closed without merging**